src/functions-and-arrays.js

```diff
@@ -1,69 +1,176 @@
+const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+function findLongestWord(arr) {
+  let word = "";
+if(arr.length === 0) {
+  return null;
+}
+else if(arr.length === 1) { 
+  return arr[0];
+}
+else {
+  for(let i=0; i<arr.length; i++) {
+    if(word.length < arr[i].length )
+    word = arr[i]
+  } 
+  return word;
+}
+}
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
+function sumNumbers(sum) {
+  let sum1 =0;
+  sum.forEach((number) => {
+      sum1 = sum1 + number;
+  })
+  return sum1;
+}
 
 
-// Iteration #3.1 Bonus:
-function sum() {}
+// // // Iteration #3.1 Bonus:
+function sum(array)  {
+  let sum1= 0;
+  if(array.length === 0){
+    return 0;
+  }
+  for(let i=0; i<array.length; i++){ 
+    
+    //should return the sum when passed array of strings:
+    if(typeof array[i] === "number"){
+      sum1 += array[i]
+  } else if(typeof array[i] === "string") {
+        sum1 += array[i].length
+  }
+  return sum1;
+  }
+
+
+let sum = 0
+for (let i=0; i<mixArr.length; i++) {
+  if (typeof mixArr[i] === "object") {
+    throw new Error;
+  }
+  else if (typeof mixArr[i] === true) {
+    mixArr[i] = 1
+  }
+  else if (typeof mixArr[i] === false) {
+    mixArr[i] = 0
+  }
+  else if (typeof mixArr[i] === "string") {
+    mixArr[i] = mixArr[i].length
+  }
+  sum+=mixArr[i]
+}
+return sum
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
+function averageNumbers(numbers) {
+  if(numbers.length === 0) return null;
+  //should return null if receives an empty array when called
+  return (sumNumbers(numbers)/ numbers.length )
+}
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
-
-// Bonus - Iteration #4.1
-function avg() {}
-
-// Iteration #5: Unique arrays
-const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
-];
-
-function uniquifyArray() {}
+function averageWordLength(arr) {
+  if(arr.length === 0) return null;
 
+  let total = 0
+  for (i=0 ; i<arr.length; i++) {
+   total += arr[i].length
+  }
+  let average = total/arr.length
+  return average
+}
 
+// // Bonus - Iteration #4.1
+// function avg() {}
+
+// // Iteration #5: Unique arrays
+// const wordsUnique = [
+//   'crab',
+//   'poison',
+//   'contagious',
+//   'simple',
+//   'bring',
+//   'sharp',
+//   'playground',
+//   'poison',
+//   'communion',
+//   'simple',
+//   'bring'
+// ];
+
+
+/* function uniquifyArray(myArr) {
+if(myArr.length === 0) return null;
+
+let newArr = [];
+for(let i=0; i< myArr.length; i++) {
+  if(!newArr.includes(myArr[i])){
+newArr.push(arr[i])
```

Comment:
Here you had everything correct, except that when you are pushing the element for the `newArr`, you misspelled `myArr[i]`.